### PR TITLE
Cleaning up AWSMobileClient code

### PR DESCRIPTION
**Pull request by royjit**
*Description of changes:*
- Moved json encode/decode to separate helper class
- Moved error related helper method to AWSMobileClientError extension
- Rearranged methods in AWSMobileClient 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AWSAuthSDK/Sources/AWSMobileClient/AWSMobileResults.swift [hunk 1]
```diff
@@ -66,9 +66,76 @@ public enum AWSMobileClientError: Error {
...
+        }
+        return error
+    }
     
-    static func CognitoAuthErrorMappingHelper(error: AWSCognitoAuthClientErrorType) -> AWSMobileClientError {
+    public static func getMobileError(cognitoAuthError: AWSCognitoAuthClientErrorType) -> AWSMobileClientError {
+        return AWSMobileClientError.CognitoAuthErrorMappingHelper(error: cognitoAuthError)
+    }
+    
+    public static func CognitoAuthErrorMappingHelper(error: AWSCognitoAuthClientErrorType) -> AWSMobileClientError {
```
Comment (palpatim) [inline AWSAuthSDK/Sources/AWSMobileClient/AWSMobileResults.swift]: Now that we're making this public, let's take the opportunity to rename it to something that doesn't sound like a class constructor. Maybe both of these could be:

```swift
public static func makeMobileClientError(from error: AWSCognitoAuthClientErrorType) -> AWSMobileClientError { ... }

public static func makeMobileClientError(fromErrorCode errorCode: String....

```

Comment (royjit) [reply AWSAuthSDK/Sources/AWSMobileClient/AWSMobileResults.swift]: Renamed, keeping it as static initializer.  `public static func makeMobileClientError(from error: Error) -> Error` returns an `Error` if not mapping is found so cannot be made into a init method.

File: AWSAuthSDK/Sources/AWSMobileClient/Internal/JSONHelper.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+//  JSONHelper.swift
+//  AWSMobileClient
+//
+
+import Foundation
+
+
```
Comment (palpatim) [inline AWSAuthSDK/Sources/AWSMobileClient/Internal/JSONHelper.swift]: nit: remove extra vertical whitespace

Comment (royjit) [reply AWSAuthSDK/Sources/AWSMobileClient/Internal/JSONHelper.swift]: Removed

**Pull request merged**